model.py

```diff
@@ -601,3 +624,167 @@ class PlaceAlias(Base):
     __table_args__ = (
         UniqueConstraint('place_id', 'name', 'language'),
     )
+
+
+class DelegatedPatronIdentifier(Base):
+    """An identifier generated by the library registry which identifies a
+    patron of one of the libraries.
+
+    This is probably an Adobe Account ID.
+    """
+    ADOBE_ACCOUNT_ID = u'Adobe Account ID'
+    
+    __tablename__ = 'delegatedpatronidentifiers'
+    id = Column(Integer, primary_key=True)
+    type = Column(String(255), index=True)
+    library_id = Column(Integer, ForeignKey('libraries.id'), index=True)
+
+    # This is the ID the foreign library gives us when referring to
+    # this patron.
+    patron_identifier = Column(String(255), index=True)
+
+    # This is the identifier we made up for the patron. This is what the
+    # foreign library is trying to look up.
+    delegated_identifier = Column(String)
+    
+    __table_args__ = (
+        UniqueConstraint('type', 'library_id', 'patron_identifier'),
+    )
+
+    @classmethod
+    def get_one_or_create(
+            cls, _db, library, patron_identifier, identifier_type,
+            create_function
+    ):
+        """Look up the delegated identifier for the given patron. If there is
+        none, create one.
+
+        :param library: The Library in charge of the patron's record.
+
+        :param patron_identifier: An identifier used by that library
+         to distinguish between this patron and others. This should be
+         an identifier created solely for the purpose of identifying
+         the patron with the library registry, and not (e.g.) the
+         patron's barcode.
+
+        :param identifier_type: The type of the delegated identifier
+         to look up. (probably ADOBE_ACCOUNT_ID)
+
+        :param create_function: If this patron does not have a
+         DelegatedPatronIdentifier, one will be created, and the given
+         function will be called to determine the value of
+         DelegatedPatronIdentifier.delegated_identifier.
+
+        :return: A 2-tuple (DelegatedPatronIdentifier, is_new)
+        """
+        identifier, is_new = get_one_or_create(
+            _db, DelegatedPatronIdentifier, library=library,
+            patron_identifier=patron_identifier, type=identifier_type
+        )
+        if is_new:
+            identifier.delegated_identifier = create_function()
+        return identifier, is_new
+
+
+class ShortClientTokenDecoder(ShortClientTokenTool):
```

Comment:
This class decodes a short client token into a DelegatedPatronIdentifier. (The circ manager version decodes it into two strings, `library_uri` and `patron_identifier`) I put it in model because I think of it as a helper class for DelegatedPatronIdentifier.